### PR TITLE
Add a specification for the logging API

### PR DESCRIPTION
**Pull request by jkwatson**
This is to provide specification to match this PR in the Java Telemetry SDK: https://github.com/newrelic/newrelic-telemetry-sdk-java/pull/159

### CODE REVIEW

File: capabilities.md [hunk 1]
```diff
@@ -157,6 +157,34 @@ Relic.
...
   An event batch is a [batch](#batches) that contains only events.
 
+## Logs
+
+The Telemetry SDK provides an API for sending log entries to
+New Relic's Log API. This implementation is _not recommended_ for sending high-throughput
+logs to New Relic. That capability should be provided by a log forwarder, rather than using
+the Telemetry SDK. The use-cases for using the implementation of the Log API in the Telemetry
+SDK should be limited to cases where using a log forwarder would not be appropriate. As an example,
+it might be appropriate to use this capability to implement the sending of OpenTelemetry Span "Events",
```
Comment (justinfoote) [inline capabilities.md:167]: It's good that you have an example here, but I don't feel very confident that I could decide whether to use this SDK to send other data to the logging API based on the guidance here.  Do I decide based on the rate of log events?

File: capabilities.md [hunk 2]
```diff
@@ -157,6 +157,34 @@ Relic.
 
   An event batch is a [batch](#batches) that contains only events.
 
+## Logs
+
+The Telemetry SDK provides an API for sending log entries to
+New Relic's Log API. This implementation is _not recommended_ for sending high-throughput
+logs to New Relic. That capability should be provided by a log forwarder, rather than using
+the Telemetry SDK. The use-cases for using the implementation of the Log API in the Telemetry
```
Comment (justinfoote) [inline capabilities.md:165]: This sentence kind of goes around in a circle.  
I think I'd strike it entirely.

File: capabilities.md (hunk 1 above)
Comment (jkwatson) [reply capabilities.md:167]: It's a good question, and I don't have a good answer. Do you have any ideas about how to provide more concrete guidance?

File: capabilities.md (hunk 2 above)
Comment (jkwatson) [reply capabilities.md:165]: which sentence, exactly?

File: capabilities.md [hunk 3]
```diff
@@ -157,6 +157,34 @@ Relic.
...
+SDK should be limited to cases where using a log forwarder would not be appropriate. As an example,
```
Comment (justinfoote) [inline capabilities.md:166]: This sentence:
```suggestion
the Telemetry SDK. As an example,
```

File: capabilities.md (hunk 1 above)
Comment (justinfoote) [reply capabilities.md:166]: OK, here are my partially-formed thoughts about why one wouldn't want to use this SDK to send application logs:
* The memory footprint needed to keep all application logs in-memory can be very high, and is unbounded.
* Using a file-tailing log forwarder is more resilient to application errors.

Comment (justinfoote) [reply capabilities.md:166]: I found this useful blog post from our logging team: https://blog.newrelic.com/engineering/best-practices-for-log-forwarding/

According to it, the downsides to sending in-process are:

1. Backpressure on HTTP requests can disrupt the normal functions of instrumented code, particularly if logs reach an unexpected size or rate.
2. When sending data to the logging backend, latencies can cause lags in the transmission of log data.
3. Network connectivity issues can lead to lost logs, which is especially troubling when agent logs from an APM-instrumented application correlate with an outage.

Could we crystalize this down to a single sentence?  Or link to this post?

Comment (jkwatson) [reply capabilities.md:166]: I like the idea of linking to the post!

Comment (jkwatson) [reply capabilities.md]: updated. let me know what you think.

**Pull request merged**